app/models/miq_group.rb

```diff
@@ -297,10 +297,11 @@ def current_user_group?
   end
 
   def ensure_can_be_destroyed
-    raise _("The login group cannot be deleted") if current_user_group?
-    raise _("The group has users assigned that do not belong to any other group") if single_group_users?
-    raise _("A tenant default group can not be deleted") if tenant_group? && referenced_by_tenant?
-    raise _("A read only group cannot be deleted.") if system_group?
+    errors.add(:base, _("The login group cannot be deleted")) if current_user_group?
```

Comment:
@yrudman 
What clears previous invocations of `errors.add`?
Would it be safer or necessary to invoke `errors.clear` before attempting to add any new errors?

Reply:
I think  it make sense to keep all messages (if it happens)

Reply:
To clarify: it make sense to keep all 3 errors if it happened.

If someone would try to execute several times `destroy` on the same instance then message will be present in `errors` several times. Should we clear before attempting to add ?  - I am not sure, I do not think there is some risk, and I did not see clearing `errors`  before

Reply:
@yrudman `If someone would try to execute several times destroy on the same instance then message will be present in errors several times.` Yes this is exactly the condition I wanted you to think about. Does it makes sense to keep adding the errors several times, for each invocation of `destroy`?